circuitbreaker.go

```diff
@@ -293,6 +304,31 @@ func (cb *Breaker) Fail() {
 	}
 }
 
+// FailWithError is the same as Fail, but keeps history of errors in internal ring buffer
+func (cb *Breaker) FailWithError(err error) {
+	cb.errorsBuffer = cb.errorsBuffer.Next()
+	cb.errorsBuffer.Value = err
+	cb.Fail()
+}
+
+// Error returns last error from internal buffer
+func (cb *Breaker) Error() error {
+	if cb.errorsBuffer.Value == nil {
+		return nil
+	}
+	return cb.errorsBuffer.Value.(error)
+}
+
+// Errors returns all errors from internal buffer
+func (cb *Breaker) Errors() (errors []error) {
+	cb.errorsBuffer.Do(func(x interface{}) {
```

Comment:
This will output the current error and then the rest.

errors ordered as `1 2 3` will be output as `3 1 2`